src/phmdoctest/cases.py

```diff
@@ -87,10 +87,9 @@ def needs_sys(blocks: List[FencedBlock]) -> bool:
 def has_pytest_mark_decorator(blocks: List[FencedBlock]) -> bool:
     """True if any code blocks generate @pytest.mark.* decorators."""
     for block in blocks:
-        if block.role == Role.CODE:
-            for directive in block.directives:
-                if directive.type in [Marker.SKIP, Marker.SKIPIF]:
-                    return True
+        any_mark = any(dr.type in (Marker.MARK_SKIP, Marker.MARK_SKIPIF) for dr in block.directives)
+        if block.role == Role.CODE and any_mark:
+            return True
```

Comment:
this is the fix core

Reply:
- Looks like Marker.SKIP_IF renamed to Marker.MARK_SKIPIF.
- Looks like no other semantic/logic changes.